pkg/virt-handler/isolation/process_test.go

```diff
@@ -0,0 +1,134 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package isolation
+
+import (
+	"github.com/mitchellh/go-ps"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var emptyProcessList = []ps.Process{}
+
+var _ = Describe("process", func() {
+	const proc1Andproc5Exec = "processA"
+	const proc2Andproc4Pid = 110
+	const nonExistPPid = 300
+
+	procStub1 := ProcessStub{ppid: 1, pid: 120, binary: proc1Andproc5Exec}
+	procStub2 := ProcessStub{ppid: proc2Andproc4Pid, pid: 2222, binary: "processB"}
+	procStub3 := ProcessStub{ppid: 1, pid: 110, binary: "processC"}
+	procStub4 := ProcessStub{ppid: proc2Andproc4Pid, pid: 3333, binary: "processD"}
+	procsStubList := []ProcessStub{procStub1, procStub3, procStub2, procStub4}
+	processes := convertToPsGoProcessList(procsStubList)
+
+	Context("find child processes", func() {
+		proc2AndProc4StubList := []ProcessStub{procStub2, procStub4}
+		proc2AndProc4 := convertToPsGoProcessList(proc2AndProc4StubList)
+
+		table.DescribeTable("should return the correct child processes of the given pid",
+			func(processes []ps.Process, ppid int, expectedProcesses []ps.Process) {
+				Expect(childProcesses(processes, ppid)).
+					To(ConsistOf(expectedProcesses))
+			},
+			table.Entry("given no input processes, there are no child processes",
+				emptyProcessList, nonExistPPid, emptyProcessList,
+			),
+			table.Entry("given process list and non-exist pid, should return no child processes",
+				processes, nonExistPPid, emptyProcessList,
+			),
+			table.Entry("given process list and pid where there are child processes of the given pid",
+				processes, proc2Andproc4Pid, proc2AndProc4,
+			),
+		)
+	})
+
+	Context("lookup process by executable", func() {
+		proc1 := convertToPsGoProcess(&procStub1)
+
+		procStub5 := ProcessStub{ppid: 100, pid: 220, binary: proc1Andproc5Exec}
+		procsListWithProcsRunningSameExec := convertToPsGoProcessList(append(procsStubList, procStub5))
+
+		table.DescribeTable("should return nil",
+			func(processes []ps.Process, executable string) {
+				Expect(lookupProcessByExecutable(processes, executable)).To(BeNil())
+			},
+			table.Entry("given no input processes and empty string as executable",
+				emptyProcessList, "",
+			),
+			table.Entry("given no input processes and executable",
+				emptyProcessList, "processA",
+			),
+			table.Entry("given processes list and empty string",
+				processes, "",
+			),
+		)
+
+		table.DescribeTable("should return the first occurrence of a process that runs the given executable",
+			func(processes []ps.Process, executable string, expectedProcess ps.Process) {
+				Expect(lookupProcessByExecutable(processes, executable)).
+					To(Equal(expectedProcess))
+			},
+			table.Entry("given processes list that includes exactly one process that runs the given executable",
```

Comment:
`runs the given executable` is repeating the text from the describe table. Please check it reads well when you run it with verbose.

Reply:
I think removing the world given makes it better
```
process
  find child processes
    should return the first occurrence of a process that runs the given executable,
      given processes list that includes exactly one process that runs the executable
```